ereuse_devicehub/resources/device/models.py

```diff
@@ -452,6 +455,23 @@ def privacy(self):
             if privacy
         )
 
+    def add_mac_to_hid(self, components_snap=None):
+        """Returns the Naming.hid with the first mac of network adapter, 
+        following an alphabetical order.
+        """
+        self.set_hid()
+        if not self.hid:
+            return
+        components = self.components if components_snap is None else components_snap
+        macs_network = [c.serial_number for c in components
+                        if c.type == 'NetworkAdapter' and c.serial_number is not None]
+        macs_network.sort()
+        mac = macs_network[0] if macs_network else ''
+        if not mac or mac in self.hid:
```

Comment:
In which situations `mac in self.hid is True`? Is to avoid that this method is called twice?

Reply:
When the mac is already in the hid. Maybe up more than one time one snapshot for one device. And then we have one device with one hid with the mac inside. If we don't put this condition we are modify the hid every time than we up one snapshot